### DISCUSSION TITLE
Serving static files in production?

### DISCUSSION THREAD

**Discussion by kalafut** [nickjj/docker-django-example#52] [Q&A]
It has been a long time since I've used Django, and I remember way back that static files were sort of a pain. I guess they still are 😀. I'm using your setup as-is and everything works fine with `DEBUG=true`, but as soon as I set it to `false` I get 404s for the static files. I did run `./run manage collectstatic` and restarted everything to no avail. FWIW I don't see many of the static files in the `public_collected` folder inside the container, I do see the admin and debug_toolbar folders (and content) though.

Any ideas?

**Comment by kalafut** [comment]
Figured it out... I also needed to set `export NODE_ENV=production` and then things worked. It might be worth documenting the relationship there in the `.env.example` comments?

**Reply by nickjj** [reply]
Thanks. Yeah the `Dockerfile` runs a collectstatic if those envs are set to production which in turn builds your assets.
